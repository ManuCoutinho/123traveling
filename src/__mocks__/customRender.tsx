import { render } from '@testing-library/react'
import { RenderOptions } from '@testing-library/react/types'
import { ChakraProvider } from '@chakra-ui/react'
import { ReactElement, FC, ReactNode } from 'react'

const MockProviders: FC<{ children: ReactNode }> = ({ children }) => {
	return <ChakraProvider>{children}</ChakraProvider>
}

const customRender = (
	ui: ReactElement,
	options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: MockProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
