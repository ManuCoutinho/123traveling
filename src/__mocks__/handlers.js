import { rest } from 'msw'

export const handlers = [
	rest.get(
		'https://api.unsplash.com/photos/nxkinK7hhOk',
		async (req, res, ctx) => {
			return res(
				ctx.status(200),
				ctx.json({
					full: 'https://images.unsplash.com/photo-1629984557780-4dde2292e245?crop=entropy\u0026cs=srgb\u0026fm=jpg\u0026ixid=MnwzMTA0MzV8MHwxfGFsbHx8fHx8fHx8fDE2ODI1OTYyODU\u0026ixlib=rb-4.0.3\u0026q=85',
					regular:
						'https://images.unsplash.com/photo-1629984557780-4dde2292e245?crop=entropy\u0026cs=tinysrgb\u0026fit=max\u0026fm=jpg\u0026ixid=MnwzMTA0MzV8MHwxfGFsbHx8fHx8fHx8fDE2ODI1OTYyODU\u0026ixlib=rb-4.0.3\u0026q=80\u0026w=1080',
					thumb:
						'https://images.unsplash.com/photo-1629984557780-4dde2292e245?crop=entropy\u0026cs=tinysrgb\u0026fit=max\u0026fm=jpg\u0026ixid=MnwzMTA0MzV8MHwxfGFsbHx8fHx8fHx8fDE2ODI1OTYyODU\u0026ixlib=rb-4.0.3\u0026q=80\u0026w=200',
					small:
						'https://images.unsplash.com/photo-1629984557780-4dde2292e245?crop=entropy\u0026cs=tinysrgb\u0026fit=max\u0026fm=jpg\u0026ixid=MnwzMTA0MzV8MHwxfGFsbHx8fHx8fHx8fDE2ODI1OTYyODU\u0026ixlib=rb-4.0.3\u0026q=80\u0026w=400',
					author: 'Bruno Thethe',
					alt: 'gray concrete bridge over river',
					userLink: 'https://unsplash.com/@bruthethe',
					downloadLocation:
						'https://api.unsplash.com/photos/nxkinK7hhOk/download?ixid=MnwzMTA0MzV8MHwxfGFsbHx8fHx8fHx8fDE2ODI1OTYyODU'
				})
			)
		}
	)
]
