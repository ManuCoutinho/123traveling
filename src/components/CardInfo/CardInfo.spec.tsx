import { screen } from '@testing-library/react'
import { render } from '__mocks__/customRender'
import { CardInfo } from '.'

const mockProps = {
	number: 'angels',
	description: 'neverland'
}

describe('<CardInfo />', () => {
	it('should render CardInfo witch correctly content', () => {
		render(<CardInfo {...mockProps}>Children</CardInfo>)
		expect(screen.getByText(/angels/i)).toBeInTheDocument()
		expect(screen.getByText(/neverland/i)).toBeInTheDocument()
		expect(screen.getByText(/children/i)).toBeInTheDocument()
	})
	it('should match to snapshot', () => {
		const { container } = render(
			<CardInfo {...mockProps}>Children</CardInfo>
		)
		expect(container.firstChild).toMatchSnapshot()
	})
})
