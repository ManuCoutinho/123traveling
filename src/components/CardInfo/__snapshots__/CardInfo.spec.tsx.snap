// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardInfo /> should match to snapshot 1`] = `
<div
  class="css-0"
>
  <p
    class="chakra-text css-k2gyu4"
  >
    angels
  </p>
  <div
    class="chakra-stack css-84zodg"
  >
    <p
      class="chakra-text css-1jtvb9"
    >
      neverland
    </p>
    <span>
      Children
    </span>
  </div>
</div>
`;
